src/NServiceBus.Transport.Msmq/MessagePump.cs

```diff
@@ -171,29 +182,38 @@ async Task InnerProcessMessages()
 
         Task ReceiveMessage()
         {
-            return TaskEx.Run(async state =>
+            // We pass a fake state to make sure we benefit from lamda delegate caching. See https://github.com/Particular/NServiceBus/issues/3884
+            var statePassedToAvoidLamdaDelegateCaching = this;
+
+            return Task.Factory.StartNew(async state =>
```

Comment:
what's the reasoning to switch to all the Factory.StartNew that we didn't use before?

Reply:
It was used before see https://github.com/Particular/NServiceBus.Transport.Msmq/pull/225/files#diff-18f43e07cacce26b57ed472dfc89a38e8bfab0fb088fa76847b547b5eb4b25fdL37

Reply:
ah, missed this 👍